addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1239,14 +1240,24 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
             </we-select>
         `)[0]);
         const recursiveFindCircular = (el) => {
-            if (el.dataset.visibilityDependency === this._getFieldName()) {
-                return true;
-            }
-            const dependencyInputEl = this._getDependencyEl(el);
-            if (!dependencyInputEl) {
-                return false;
+            const dependentFieldName = this._getFieldName(el);
+            // Get all the fields that have the same label than the input.
+            let fieldEls = Array.from(this.formEl
+                .querySelectorAll(`.s_website_form_input[name="${dependentFieldName}"]`))
+                .map((el) => el.closest(".s_website_form_field"));
+            // Filter the array to remove the duplicate fields. This could
+            // happen if the field has multiple inputs ("Multiple Checkboxes" for
+            // example.)
+            fieldEls = fieldEls.filter((fieldEl, index) => fieldEls.indexOf(fieldEl) === index);
+            const fieldName = this._getFieldName();
+            for (const fieldEl of fieldEls) {
+                const dependencyInputEl = this._getDependencyEl(fieldEl);
```

Comment:
You don't need to do `_getDependencyEl` if `fieldEl.dataset.visibilityDependency === fieldName` so you could avoid to do it.